src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -173,3 +173,110 @@ def watch():
         t = threading.Thread(target=watch, daemon=True)
         t.start()
         t.join()
+
+
+class ZFSSnapshot(Service):
+
+    class Config:
+        namespace = 'zfs.snapshot'
+
+    @accepts(
+        Str('dataset'),
+        Str('name'),
+        Bool('recursive'),
+        Int('vmsnaps_count')
+    )
+    def create(self, dataset, name, recursive=False, vmsnaps_count=0):
```

Comment:
But it should accept a dict as an argument, to keep the same API of CRUDServicec.